#328 Add the last remaining user fields from current BugGuide.

kleintom opened the pull request:
field_user_city_state_country base was added in https://github.com/bugguide/bugguide/commit/357103e74353424de7874a95eda81fefc60b68fb; it's already being migrated. Here we add the instance definition and display it on a user's profile page.

We also add the default state/province and signature fields.

Note to self: as with other textarea user fields, we've deleted the text formats settings section on the signature field (ignore the actual settings here, this was from an early attempt):
      'better_formats' => array(
        'allowed_formats' => array(
          'f' => 'f',
          'php_code' => 0,
          'plain_text' => 0,
          'u' => 0,
          'xbbcode' => 'xbbcode',
        ),
        'allowed_formats_toggle' => 0,
        'default_order_toggle' => 0,
        'default_order_wrapper' => array(
          'formats' => array(
            'f' => array(
              'weight' => -1,
            ),
            'php_code' => array(
              'weight' => 11,
            ),
            'plain_text' => array(
              'weight' => 10,
            ),
            'u' => array(
              'weight' => 1,
            ),
            'xbbcode' => array(
              'weight' => -6,
            ),
          ),
        ),
      ),
If you go to admin/config/people/accounts/fields/field_user_signature, the 'Text Processing' section is set to 'Filtered text (user selects text format)' - that's the 'text_processing' => 1 setting - and then I guess we're just relying on the text format settings at admin/config/content/formats to determine which users see which formats (since we're not restricting any text formats in the field settings themselves).

clouseau commented:
We are leaving Signature behind. It was originally designed to leave your signature on each comment you made. In practice it is not used. Do you mind removing it from the PR?

It is a shame that city/state/country was not done as separate fields instead of a freeform text field. I will resist the urge to parse these 140,000 records during migration and will just pipe down.

kleintom commented:
Haha, I won't say it's not too late :-)

I removed the signature field, and noticed that I hadn't updated the weights on the total contributed images/comments fields, so I added that change as well (they're special ds fields whose weights are set in a different place than the others; everything was still working, but two fields wound up with the same weight and I think I got 'lucky' with the alphabetical order of those overlapping fields being the order I wanted [is that how that works??] - now all the weights are different, and in the correct order).

clouseau commented:
Yes, typically in Drupal things are sorted by weight and then alphabetically.
